### PR TITLE
Add ability to make followup message ephemeral

### PR DESCRIPTION
**Pull request by IDoEverything**
# Summary
Created a `DiscordFollowupMessageBuilder` to be able to make followup message ephemeral.

# Details
Also added the message flags for ephemeral and loading states.

### CODE REVIEW

File: DSharpPlus/Entities/Interaction/DiscordFollowupMessageBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        /// </summary>
+        /// <param name="ephemeral">Ephemeral.</param>
+        public DiscordFollowupMessageBuilder AsEphemeral(bool ephemeral)
+        {
+            this.IsEphemeral = ephemeral;
+            return this;
+        }
+
+        /// <summary>
+        /// Allows for clearing the Webhook Builder so that it can be used again to send a new message.
```
Comment (Neuheit) [inline DSharpPlus/Entities/Interaction/DiscordFollowupMessageBuilder.cs:212]: Typo here

**Comment by Neuheit** [comment]
Also, did we want to consider having this just inherit from DiscordMessageBuilder or DiscordWebhookBuilder?

**Comment by IDoEverything** [comment]
> Also, did we want to consider having this just inherit from DiscordMessageBuilder or DiscordWebhookBuilder?

I completely forgot about inheritance, though that would mess up the descriptions

**Comment by Neuheit** [comment]
I think if we could get this to inherit from the existing message builder that would be ideal.

**Comment by Neuheit** [comment]
Nvm on my last statement, I'll rework this into Glock's PR.

**Review by Neuheit — APPROVED** [review]

**Pull request merged**